test/test_unitary.py

```diff
@@ -84,3 +84,38 @@ def test_compute_gates(self):
         calc2 = v_gate @ np.diag(diag).conj().T @ w_gate
         self.assertTrue(np.allclose(calc1, gate1))
         self.assertTrue(np.allclose(calc2, gate2))
+
+    def test_compute_gates_fixed(self):
+        """ test auxiliar funciont compute gates"""
+        # Matrices gate1 and gate2 are not unitary.
+        gate1 = np.array(
+                [[ 1.39499204e-01-1.02062065e-01j, -3.73248642e-01-6.00570938e-01j,
+                   5.83554297e-01-1.84683236e-01j, -2.69576508e-01+1.51024165e-01j],
+                 [-4.71845544e-01-6.91006095e-01j, -7.85046229e-17+0.00000000e+00j,
+                   2.78615481e-01+1.87070569e-01j,  4.00199035e-01-1.64593766e-01j],
+                 [ 7.04556392e-02-4.84927805e-01j,  0.00000000e+00+9.81307787e-18j,
+                  -3.29381826e-01-1.72042433e-01j, -6.36139910e-01-4.65957139e-01j],
+                 [-1.02062065e-01-1.39499204e-01j,  6.00570938e-01-3.73248642e-01j,
+                  -1.84683236e-01-5.83554297e-01j,  1.51024165e-01+2.69576508e-01j]])
+        gate2 = np.array(
+                [[ 1.72752243e-01-5.77541244e-03j,  2.97965199e-02-7.06478710e-01j,
+                   5.86222022e-01+1.76031973e-01j, -3.07814277e-01-2.70215300e-02j],
+                 [-8.22782535e-01+1.52172736e-01j,  4.11897902e-17+7.20821328e-17j,
+                   2.83747390e-01-1.79190967e-01j,  1.12670242e-02-4.32577657e-01j],
+                 [-4.15037537e-01-2.60504925e-01j, -1.02974475e-17+4.11897902e-17j,
+                  -2.90528108e-01+2.31698952e-01j, -6.83375688e-01+3.93430693e-01j],
+                 [-5.77541244e-03-1.72752243e-01j,  7.06478710e-01+2.97965199e-02j,
+                   1.76031973e-01-5.86222022e-01j, -2.70215300e-02+3.07814277e-01j]])
+
+        # Let's find the closest unitary matrices to gate1 and gate2.
+        def closest_unitary(matrix):
+            svd_u,_,svd_v = np.linalg.svd(matrix)
+            return svd_u.dot(svd_v)
+        gate1 = closest_unitary(gate1)
+        gate2 = closest_unitary(gate2)
```

Comment:
```suggestion

```

This new `_compute_gates` already checks this problem.

Reply:
I will remove this test. It is no longer necessary.